pkg/querier/worker/worker.go

```diff
@@ -150,20 +150,8 @@ func NewQuerierWorker(cfg Config, handler RequestHandler, log log.Logger, reg pr
 		grpcCfg = cfg.QuerySchedulerGRPCClientConfig
 		workerClient = "query-scheduler-worker"
 		processor, servs = newSchedulerProcessor(cfg, handler, log, reg)
-
-	case cfg.FrontendAddress != "":
-		level.Info(log).Log("msg", "Starting querier worker connected to query-frontend", "frontend", cfg.FrontendAddress)
-
-		factory = func(receiver servicediscovery.Notifications) (services.Service, error) {
-			return servicediscovery.NewDNS(log, cfg.FrontendAddress, cfg.DNSLookupPeriod, receiver)
-		}
-
-		grpcCfg = cfg.QueryFrontendGRPCClientConfig
-		workerClient = "query-frontend-worker"
-		processor = newFrontendProcessor(cfg, handler, log)
-
 	default:
-		return nil, errors.New("no query-scheduler or query-frontend address")
+		return nil, errors.New("no query-scheduler address")
```

Comment:
Should we turn this into something like `if !cfg.IsSchedulerConfigured() { ... }` before the `switch`? Then we could drop the `switch` altogether.